src/Bicep.Core/Semantics/SymbolExtensions.cs

```diff
@@ -25,5 +27,17 @@ public static SyntaxBase UnsafeGetBodyPropertyValue(this ResourceSymbol resource
 
         public static SyntaxBase? SafeGetBodyPropertyValue(this ModuleSymbol moduleSymbol, string propertyName)
             => SafeGetBodyProperty(moduleSymbol, propertyName)?.Value;
+
+        public static bool IsSecure(this ParameterSymbol parameterSymbol)
+        {
+            // local function
+            bool isSecure(DecoratorSyntax? value) => value?.Expression is FunctionCallSyntax functionCallSyntax && functionCallSyntax.NameEquals("secure");
+
+            if (parameterSymbol?.DeclaringSyntax is ParameterDeclarationSyntax paramDeclaration)
+            {
+                return paramDeclaration.Decorators.Any(d => isSecure(d));
```

Comment:
>Decorators [](http://example.com/codeflow?start=40&length=10)

We also have the deprecated modifier syntax to mark something as secure. We should probably handle it here as well.

Reply:
@alex-frankel need to talk about length of support for this.

Reply:
decision is not to support the deprecated syntax in this version of the linter